crates/router/src/utils.rs

```diff
@@ -155,3 +158,51 @@ pub fn to_currency_base_unit_asf64(
 pub fn get_payment_attempt_id(payment_id: impl std::fmt::Display, attempt_count: i16) -> String {
     format!("{payment_id}_{attempt_count}")
 }
+
+#[derive(Debug)]
+pub struct QrImage;
+
+pub trait CreateQrImageUrlFromQrData {
+    fn create_image_data_source_url(
+        &self,
+        _qr_code_data: String,
+    ) -> Result<String, error_stack::Report<common_utils::errors::QrCodeError>>;
+}
+
+impl CreateQrImageUrlFromQrData for QrImage {
+    fn create_image_data_source_url(
+        &self,
+        qr_code_data: String,
+    ) -> Result<String, error_stack::Report<common_utils::errors::QrCodeError>> {
+        let qr_code = qrcode::QrCode::new(qr_code_data.as_bytes())
+            .into_report()
+            .change_context(common_utils::errors::QrCodeError::FailedToCreateQrCode)?;
+
+        // Renders the QR code into an image.
+        let qrcode_image_buffer = qr_code.render::<Luma<u8>>().build();
+        let qrcode_dynamic_image = image::DynamicImage::ImageLuma8(qrcode_image_buffer);
+
+        let mut image_bytes: Vec<u8> = Vec::new();
+
+        // Encodes qrcode_dynamic_image and write it to image_bytes
+        let _ = qrcode_dynamic_image.write_to(&mut image_bytes, image::ImageOutputFormat::Png);
```

Comment:
You can use `BufWriter` here if in the future if you do two or more writes this is suggested.

Reply:
There is no `BufWriter` function implemented for `DynamicImage`

Reply:
Not needed